nwbinspector/register_checks.py

```diff
@@ -0,0 +1,139 @@
+"""Authors: Cody Baker and Ben Dichter."""
+from collections import defaultdict, OrderedDict
+from functools import wraps
+from enum import Enum
+from dataclasses import dataclass
+
+import h5py
+
+
+class Importance(Enum):
+    """A definition of the valid importance levels for a given check function."""
+
+    CRITICAL = 2
+    BEST_PRACTICE_VIOLATION = 1
+    BEST_PRACTICE_SUGGESTION = 0
+
+
+class Severity(Enum):
+    """
+    A definition of the valid severity levels for the output from a given check function.
+
+    Strictly for internal development that improves report organization; users should never directly see these values.
+    """
+
+    HIGH = 2
+    LOW = 1
+    NO_SEVERITY = 0
+
+
+global available_checks
+available_checks = OrderedDict({importance: defaultdict(list) for importance in Importance})
+
+
+@dataclass
+class InspectorMessage:
+    """
+    The primary output to be returned by every check function.
+
+    Parameters
+    ----------
+    message : str
+        A message that informs the user of the violation.
+    severity : Severity, optional
+        If a check of non-CRITICAL importance has some basis of comparison, such as magitude of affected data, then
+        the developer of the check may set the severity as Severity.HIGH or Severity.LOW by calling
+        `from nwbinspector.register_checks import Severity`. A good example is comparing if h5py.Dataset compression
+        has been enabled on smaller vs. larger objects (see nwbinspect/checks/nwb_containers.py for details).
+
+        The user will never directly see this severity, but it will prioritize the order in which check results are
+        presented by the NWBInspector.
```

Comment:
It would be useful to add documentation for the other parameters as well.

Reply:
I didn't do that originally as I did not want the user to think they had to or even should set those themselves at `__init__` of the object.

I could break up the docstring into `Parameters` and `Returns` and add the documentation details for the other fields under `Returns`, thus implying that all those others values are things that should be 'returned' or otherwise filled in automatically by the class after the check completes.